consensus/istanbul/backend/engine_test.go

```diff
@@ -812,3 +821,184 @@ func TestSnapshot(t *testing.T) {
+				{vote{"istanbul.committeesize", uint64(22)}, 11},
+				{vote{"istanbul.committeesize", uint64(22)}, 12},
+				{vote{"istanbul.committeesize", uint64(22)}, 13},
+				{vote{"istanbul.committeesize", uint64(22)}, 14},
+				{vote{"istanbul.committeesize", uint64(22)}, 0}, // check on current
+
+				// unitprice for all blocks
+				{vote{"governance.unitprice", uint64(1)}, 1},
+				{vote{"governance.unitprice", uint64(1)}, 2},
+				{vote{"governance.unitprice", uint64(1)}, 3},
+				{vote{"governance.unitprice", uint64(1)}, 4},
+				{vote{"governance.unitprice", uint64(1)}, 5},
+				{vote{"governance.unitprice", uint64(1)}, 6},
+				{vote{"governance.unitprice", uint64(1)}, 7},
+				{vote{"governance.unitprice", uint64(1)}, 8},
+				{vote{"governance.unitprice", uint64(2000000)}, 9},
+				{vote{"governance.unitprice", uint64(2000000)}, 10},
+				{vote{"governance.unitprice", uint64(2000000)}, 11},
+				{vote{"governance.unitprice", uint64(2)}, 12},
+				{vote{"governance.unitprice", uint64(2)}, 13},
+				{vote{"governance.unitprice", uint64(2)}, 14},
+				{vote{"governance.unitprice", uint64(2)}, 0}, // check on current
+			},
+		},
+	}
+
+	var configItems []interface{}
+	configItems = append(configItems, proposerPolicy(params.WeightedRandom))
+	configItems = append(configItems, epoch(3))
+	configItems = append(configItems, governanceMode("single"))
+	configItems = append(configItems, blockPeriod(0)) // set block period to 0 to prevent creating future block
+	for _, tc := range testcases {
+		chain, engine := newBlockChain(1, configItems...)
+
+		// test initial governance items
+		assert.Equal(t, uint64(3), engine.governance.Epoch())
+		assert.Equal(t, "single", engine.governance.GovernanceMode())
+		assert.Equal(t, uint64(21), engine.governance.CommitteeSize())
+		assert.Equal(t, uint64(1), engine.governance.UnitPrice())
+		assert.Equal(t, "0", engine.governance.MintingAmount())
+		assert.Equal(t, "100/0/0", engine.governance.Ratio())
+		assert.Equal(t, false, engine.governance.UseGiniCoeff())
+		assert.Equal(t, "2000000", engine.governance.MinimumStake())
+
+		// add votes and insert voted blocks
+		var (
+			previousBlock, currentBlock *types.Block = nil, chain.Genesis()
+			err                         error
+		)
+
+		for _, v := range tc.votes {
+			engine.governance.AddVote(v.key, v.value)
+			previousBlock = currentBlock
+			currentBlock = makeBlockWithSeal(chain, engine, previousBlock)
+			_, err = chain.InsertChain(types.Blocks{currentBlock})
+			assert.NoError(t, err)
+		}
+
+		// insert blocks until the vote is applied
+		for i := 0; i < 6; i++ {
+			previousBlock = currentBlock
+			currentBlock = makeBlockWithSeal(chain, engine, previousBlock)
+			_, err = chain.InsertChain(types.Blocks{currentBlock})
+			assert.NoError(t, err)
+		}
+
+		for _, item := range tc.expected {
+			blockNumber := item.appliedBlockNumber
+			if blockNumber == 0 {
+				blockNumber = chain.CurrentBlock().NumberU64()
+			}
```

Comment:
It's not a general way to indicate the latest block number. You better use `LatestBlockNumber` or `-1`. 
Pleaes, fix this later

Reply:
I will update this later. Thanks.